core/src/main/scala/com/nvidia/spark/rapids/tool/analysis/AppSQLPlanAnalyzer.scala

```diff
@@ -226,53 +264,26 @@ class AppSQLPlanAnalyzer(app: AppBase)
     }
     // Then process SQL plan metric type
     for (metric <- node.metrics) {
-      val stages =
-        sqlPlanNodeIdToStageIds.getOrElse((visitor.sqlPIGEntry.sqlID, node.id), Set.empty)
-      val allMetric = SQLMetricInfoCase(visitor.sqlPIGEntry.sqlID, metric.name,
+      val stages = getNodeStages(visitor.sqlPIGEntry.id, node.id).getOrElse(Set())
+      val allMetric = SQLMetricInfoCase(visitor.sqlPIGEntry.id, metric.name,
         metric.accumulatorId, metric.metricType, node.id, node.name, node.desc, stages)
-
       allSQLMetrics += allMetric
-      if (app.sqlPlanMetricsAdaptive.nonEmpty) {
-        val adaptive = app.sqlPlanMetricsAdaptive.filter { adaptiveMetric =>
-          adaptiveMetric.sqlID == visitor.sqlPIGEntry.sqlID &&
-            adaptiveMetric.accumulatorId == metric.accumulatorId
-        }
-        adaptive.foreach { adaptiveMetric =>
-          val allMetric = SQLMetricInfoCase(visitor.sqlPIGEntry.sqlID, adaptiveMetric.name,
-            adaptiveMetric.accumulatorId, adaptiveMetric.metricType, node.id,
-            node.name, node.desc, stages)
-          // could make this more efficient but seems ok for now
-          val exists = allSQLMetrics.filter { a =>
-            ((a.accumulatorId == adaptiveMetric.accumulatorId)
-              && (a.sqlID == visitor.sqlPIGEntry.sqlID)
-              && (a.nodeID == node.id && adaptiveMetric.metricType == a.metricType))
-          }
-          if (exists.isEmpty) {
-            allSQLMetrics += allMetric
-          }
-        }
-      }
```

Comment:
This entire block was useless because the condition never is realized. The condition that the adaptiveAccumId already existed was true all the time and therefore, it never added any value. For some eventlogs, this block would cost significant CPUTime depending on the number of AdaptiveAQEMetric updates in the eventlogs.